zkevm-circuits/src/evm_circuit/util/constraint_builder.rs

```diff
@@ -0,0 +1,519 @@
+use crate::{
+    evm_circuit::{
+        step::{ExecutionResult, Preset, Step},
+        table::{FixedTableTag, Lookup, RwTableTag, TxContextFieldTag},
+        util::{Cell, Word},
+    },
+    util::Expr,
+};
+use halo2::{arithmetic::FieldExt, plonk::Expression};
+use std::convert::TryInto;
+
+// Max degree allowed in all expressions passing through the ConstraintBuilder.
+const MAX_DEGREE: usize = 2usize.pow(3) + 1 + 32;
```

Comment:
`MAX_DEGREE` is allowed more with `32` since the randomness now is from a single instance column, tho it's a constant polynomial, which should not increase polys' degree in theorem, but halo2 counts it anyway.

But if we allocate 32 instance columns, I'm afraid it would affect a lot on the benchmark of an actual prover, so here I tried to just allow it to be so large.